Sources/TuistGenerator/Models/BuildConfiguration.swift

```diff
@@ -1,15 +1,50 @@
 import Foundation
 
-public enum BuildConfiguration: String {
-    case debug
-    case release
+public struct BuildConfiguration {
+    public enum Variant: String {
+        case debug, release
+    }
+
+    public static let debug = BuildConfiguration(name: "Debug", predefined: true, variant: .debug)
+    public static let release = BuildConfiguration(name: "Release", predefined: true, variant: .release)
+
+    public let name: String
+    public let variant: Variant
+
+    private let predefined: Bool
```

Comment:
What's this attribute used for?

Reply:
I put it only for debugging purposes, it does not play any role except this tint bit of extra information if the Configuration was created "manually" by the client code or it's one of the 2 predefined (`.debug` and `.release`) - that's it. I am happy to remove it.